macro/vignettes/pfSI.Rmd

```diff
@@ -0,0 +1,385 @@
+* Predictable Variables
+  + biting weight - this represents the amount of skin this person has,
+    so it changes with age. Lowered by PE vaccination
+  + kappa - if the person is infected and _not in a resevoir,_ this is c x (biting weight).
+    Lowered by GS vaccination by a factor c = c x (1 - gs_block_pf).
+  + EIR - the rate the person is getting bitten. For someone in a resevoir,
+    this is the resevoir EIR. For someone not in a resevoir, this
+    is max(0, beta x biting_weight / patch_biting_weight).
+  + incidence - the number of infectious people in a patch. This is the sum of all infectious.
+
+
+## Event Space
+
+Here, variable names are sometimes shortened. Mean is mn, and variance is vr.
+There is a notation for events that says what state you leave and what
+you enter. The minus is leave and the + is enter.
+
+* Infection
+  + infection - from state L to I (-L, +I), exponential with rate duration
+  + clearance - from state I to S (-I, +S), exponential with rate duration_pf
+  + fever - (-I, +F, +I), exponential with rate mnFever, which is incomplete by U(0, 1) < fever_pf.
+  + treatment - (-I, +T, +I) rate mnTreatPF
+  + recover - (-I, +S). This is also called clearance.
+* Chemoprophylaxis
+  + treatment - seek treatment, (-I, +P)
+  + chemoprophylaxis - duration of prophylaxis, (-P, +S), rate mnChemoprophylaxisPf
+* Vaccination
+  + PE - sporozooite-blocking vaccination, incomplete by U(0, 1) < pe_protect_pf.
+    Lowers biting weight, b.
+  + PE with treatment - same as PE, but it starts chemoprophylaxis.
+  + GS - gametocyte-blocking vaccination. incomplete by U(0, 1) < gs_protect_pf. Lowers
+    c, the probability to infect mosquitoes.
+  + GS with treatment.
+  + wane PE - (-PE, +S) rate normal(mnPE_PF, vr_PE_Pf). Sets b back to full power.
+  + wane GS - (-GS, +S) rate normal(mnGE_PF, vr_GE_Pf). Sets c back to full power.
+* Bites - These will move to another module, the biting module.
+  + The biting algorithm returns a number of bites for this person.
+  + The biting hazard is incomplete. There is a probability draw that
+    U(0, 1) < b to create move the person to the latent state.
+
+
+## Class of problems
+
+The pfSI described above fits into a particularly restrictive class of models
+because its state space is discrete and entirely Markovian.
+We have examples of simulations we'd like to write that don't fit those
+requirements. Let's look at some examples.
+
+Consider the case of assigning infectious bites to a person.
+We would like to draw the probability that a bite is infectious and the
+queue the event only if it is infectious. The way this is done for a Markov process
+is to specify the distribution of bite times as an incomplete distribution,
+meaning the maximum cumulative probability is the binomial probability that
+the bite is infectious. Maybe that's harder to figure out than thinking of it
+as two draws.
+
+Another case is when a change of state can lead to one of multiple transitions.
```

Comment:
This also seems Markov to me? Sampling the p(i->j) PMF and then the holding time density K(\tau | i->j) ?